src/crystal/system/thread.cr

```diff
@@ -1,40 +1,138 @@
+# :nodoc:
+module Crystal::System::Thread
+  # alias Handle
+
+  # def self.new_handle(thread_obj : ::Thread) : Handle
+
+  # def self.current_handle : Handle
+
+  # def self.yield_current : Nil
+
+  # def self.current_thread : ::Thread
+
+  # def self.current_thread=(thread : ::Thread)
+
+  # private def system_join : Exception?
+
+  # private def system_close
+
+  # private def stack_address : Void*
+end
+
+{% if flag?(:wasi) %}
+  require "./wasi/thread"
+{% elsif flag?(:unix) %}
+  require "./unix/pthread"
+{% elsif flag?(:win32) %}
+  require "./win32/thread"
+{% else %}
+  {% raise "Thread not supported" %}
+{% end %}
+
 # :nodoc:
 class Thread
```

Comment:
This class is not in the `Crystal::System` namespace so I don't think it belongs in `crystal/system`.
We should probably move the implementation to `thread.cr`. WDYT?

Reply:
Judging from #7995, I think there is no problem if we stay this way until we figure out the public multithreading APIs. Personally I'd prefer renaming the class to `Crystal::Thread` instead